examples/schedule_sketch.py

```diff
@@ -1,4 +1,90 @@
 
+# Limitations of current scheduling front end and the issues that needs to be resoved
+# 1. How to return statement with proc from a scheduling operator?
+#    e.g., we want to return stmt_gap (or stmt) from simple_fission_after, so that we can
+#          further fission or remove loop using that stmt
+# 2. How to pattern match / point to a statement block (instead of just a single stmt)?
+#      Currently, our pointing scheme is not very suitable for expressing statement blocks.
+#      We need to point to statement blocks with new stage_memory directive.
+# 3. How to modify the pointer with respect to the statement, by gap value or line number?
+#    e.g., We want to be able to say "this statement after that statement" (such as "g+1")
+
+# To motivate why we need 3, the previous fission_after semantics was this:
+for i in seq(0, n):
+  s1
+  s2
+----->
+for i in seq(0, n): # This is "pre"
+  s1
+for i in seq(0, n): # "post"
+  s2
+.remove_loop('for i in _:_ #0')
+.remove_loop('for i in _:_ #1')
+# To construct user-defined scheduling function like this from primitive operators
+# such as fission_simple and remove_loop, notice that we need to have a systematic way of saying
+# '... #0' and '... #1', instead of just giving a string.
+# In another words, I think we need a way for a user to address statement order or a line number.
+
+# -----------------------------------------------------------------------------------------
+# Proposal 1 is to record the previously added statement in proc.
+# Example code:
+@proc
+def foo():
+    ...
+foo = foo.simple_fission('a = b[_]') # We don't need to modify the existing interface
+
+def fission(proc, stmt, n_lifts): # Users can write user-defined schedule in Python
+    for i in range(0, n_lifts):
+        # I think we will need to have branch like this, if our semantics is:
+        # 1. If scheduling operator is called with a pattern, use that pattern
+        # 2. If the operator is called without a pattern, use the recorded stmt in proc
+        if i == 0:
+            proc = proc.simple_fission(stmt)
+        else:
+            proc = proc.simple_fission()
+
+# -----------------------------------------------------------------------------------------
+# Proposal 2 is to return the tuple of added statement and proc from scheduling directives.
+# Example code:
+def fission(proc, stmt, n_lifts):
+    # I guess this is more explicit, but can confuse users
+    for i in range(0, n_lifts):
+        (proc, stmt) = proc.simple_fission(stmt)
```

Comment:
```
for j:
  for i in seq(0, n):
    s1 <--- stmt
    s2
----------->
for j:
  for i in seq(0, n): <---- stmt
    s1
  for i in seq(0, n):
    s2
----------->
for j: <---- stmt
  for i in seq(0, n):
    s1
for j:
  for i in seq(0, n):
    s2
```

Reply:
Comment from Alex: we should think about how to handle "outdated" pointers. Should they be invalid after the first use? Should "pointers" hold data structure other than a mere object (such as traversal path) to make it re-useable?